masschallenge/django-accelerator
# AC-6380: Add Fields to the Custom Tree model and the TreeItem models

#### Changes introduced in [AC-6380](https://masschallenge.atlassian.net/browse/AC-6380)
- remove name field from base_nav_tree
- remove name field from base_nav_tree_item model and add
  permission fields
- add nav_tree and nav_tree_item factories
- create nav_tree context
- add migration file

#### How to test
- checkout AC-6380 on accelerate and django-accelerator
- visit the admin interface i.e. localhost:8181/admin
- add a [NavTree](http://localhost:8181/admin/mc/navtree) (keep note of your tree **alias**)
- select the nav tree you created, there will be a link to create a nav tree item (i.e. **ADD SITE TREE ITEM**)
- create a nav tree item, and give it a user role that you have, the following fields would do
   - title: sante
   - url: /sante
   - User role: <the-user-role-you-have>

- edit an existing template e.g. mcproject/mc_apply/templates/mc_apply/applicant.html` and add this tag at the top `{% load sitetree %}` and this `{% sitetree_menu from "<your-tree-alias>" include "trunk" %}` somewhere in the template to display the menu
- if you edited the applicant.html file, open `localhost:8181/applicant` to see the rendered menu
- now change the nav tree item and give it a user role that you don't have
- visit `localhost:8181/applicant` again and notice your tree item does not show
- feel free to play around with the different permission fields

#### Note
- this **must** be merged first before it's [sibling PR](https://github.com/masschallenge/accelerate/pull/2077)
